subprojects/model-core/src/testFixtures/groovy/org/gradle/internal/reflect/validation/ValidationMessageChecker.groovy

```diff
@@ -815,12 +828,12 @@ trait ValidationMessageChecker {
         }
 
         AnnotationContext forTransformParameters() {
-            validAnnotations = "@Console, @Inject, @Input, @InputDirectory, @InputFile, @InputFiles, @Internal, @Nested or @ReplacedBy"
+            validAnnotations = "@Console, @Inject, @Input, @InputDirectory, @InputFile, @InputFiles, @Internal, @Nested, @ReplacedBy or @ServiceReference"
```

Comment:
Will add or remove this

Reply:
I added, but we get a nag because we don't look into transformer parameter properties for service references. 

A test has ben added for this, and for the case a service has a service reference as one of its parameters, which also leads to a nag.